### ISSUE TITLE
Black Screen

### ISSUE THREAD

**Issue by springware** [appknox/droid-VNC-server#12]
I have some trouble on this droidvncsever.
I could able to build and run apk. but get black screen on realvnc viewer(from pc).
I changed the Display Access Method in Setting but all are same. black screen or server not started.
But Mouse Event is still working well. just screen has problem.
I need to make it work for almost all type of Phone.
Can anyone help me?

**Comment by subho007** [comment]
@springware you need to disable selinux. To do that you can `adb shell` into the device and run `setenforce 0`

**Comment by mixaz** [comment]
@subho007 I'm afraid you closed the issue too early. I have the same issue on Lenovo Tab4 TB-8504F, Android 7.1.1, Adreno 308 GPU. I tested both droid-vnc binary and some frame buffer examples like ones from AOSP source code https://android.googlesource.com/platform/system/extras/+/refs/tags/android-7.1.1_r58/tests/framebuffer/ and https://stackoverflow.com/a/25519470/1028256. 

I was able to use only flinger method, after fixing a bug there with segmentation fault in incorrect free().

I'm still investigating if framebuffer method can be fixed to work on this device, but apparently it doesn't work in current droid-vnc code. I'm going to test original https://github.com/LibVNC/libvncserver/blob/master/examples/androidvncserver.c as well.

... and I tested in permissive SELinux mode too
